### ISSUE TITLE
Convert String Format in ColorPicker

### ISSUE THREAD

**Issue by robloo** [unoplatform/uno#6730]
## Current behavior

The ColorPicker (and other WinUI sourced controls) currently uses strings copied verbatim from WinUI. This is a problem as discussed here https://github.com/unoplatform/uno/pull/3955#issuecomment-687243929:

> The string formatting methods can take arguments to automatically insert the specified arguments into the string. In C# this might look like "The value is {0:X2}" -> "The value is FF" in C++/WinRT the string formatting is different. Instead I have localized strings that look like:

https://github.com/unoplatform/uno/blob/938e88637a5b140b647eb688445eb255a4d786ef/src/Uno.UI/Microsoft/UI/Xaml/Controls/ColorPicker/Strings/en-us/Resources.resw#L216-L219

There already exists a method in Uno to solve this problem according to @MartinZikmund 

 > @robloo turns out this is in fact already handled in Uno :-D - Uno.UI.Helpers.WinUI.StringUtil.cs

https://github.com/unoplatform/uno/pull/4427#discussion_r516599289

String formatting code needs to be modified to pass the raw resource strings through the conversion method before string.Format in C#.

Relates to discussion in #5992 

## Expected behavior

String formats should automatically converted to a C# string.Format() compatible version.

## Environment

Nuget Package:

- [ ] Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia
- [ ] Uno.WinUI / Uno.WinUI.WebAssembly / Uno.WinUI.Skia
- [ ] Uno.SourceGenerationTasks
- [ ] Uno.UI.RemoteControl / Uno.WinUI.RemoteControl
- [ ] Other: 

Nuget Package Version(s):

Affected platform(s):

- [ ] iOS
- [ ] Android
- [ ] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [ ] macOS
- [ ] Skia
  - [ ] WPF
  - [ ] GTK (Linux)
  - [ ] Tizen
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

IDE:

- [ ] Visual Studio 2017 (version: )
- [ ] Visual Studio 2019 (version: )
- [ ] Visual Studio for Mac (version: )
- [ ] Rider Windows (version: )
- [ ] Rider macOS (version: )
- [ ] Visual Studio Code (version: )

Relevant plugins:

- [ ] Resharper (version: )

**Comment by robloo** [comment]
This one can be assigned to me. However, it will take a bit before I get to it.

@Xiaoy312  FYI

**Comment by Youssef1313** [comment]
Does it worth writing an analyzer to detect these?

**Comment by jeromelaban** [comment]
@Youssef1313 not necessarily. We don't want to change the strings, only make sure that we're converting them properly.

**Comment by Youssef1313** [comment]
Yeah that's what I meant. The analyzer would flag calls to `string.Format` where the passed parameter contains a match to this regex: `\%(\d+)!.*?!`

**Comment by jeromelaban** [comment]
It's unlikely that such a pattern will happen statically, as the string will always come from resources strings.
